matchbox_socket/src/webrtc_socket/socket.rs

```diff
@@ -301,8 +302,10 @@ impl WebRtcSocket {
     }
 
     /// Returns the id of this peer
-    pub fn id(&self) -> &PeerId {
-        &self.id
+    pub fn get_id(&self) -> Result<Option<PeerId>, UnretrievablePeerId> {
+        self.id
+            .try_lock()
+            .map_or(Err(UnretrievablePeerId), |id_mutex| Ok((*id_mutex).clone()))
```

Comment:
Is this if the mutex is poisoned? If that's the case, wouldn't that mean something is seriously wrong and we might as well panic?

Or maybe I'm mistaken? It's a bit late here :P

Reply:
Yeah, you're probably right here, will be going away in a minute anyhow - then I'm off to bed

Reply:
Good idea will do the same! Nice work :)